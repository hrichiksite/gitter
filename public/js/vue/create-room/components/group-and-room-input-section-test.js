const mount = require('../../__test__/vuex-mount');
const { default: GroupAndRoomInputSection } = require('./group-and-room-input-section.vue');

describe('GroupAndRoomInputSection', () => {
  it('matches snapshot', () => {
    const { wrapper } = mount(GroupAndRoomInputSection);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('errors matches snapshot', () => {
    const { wrapper } = mount(GroupAndRoomInputSection, {}, store => {
      store.state.createRoom.groupError = 'group error';
      store.state.createRoom.roomNameError = 'room name error';
      store.state.createRoom.groupsRequest.error = new Error('groupsRequest error');
      store.state.createRoom.reposRequest.error = new Error('reposRequest error');
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('roomName', () => {
    it('Inputing new room name text fires store mutation', () => {
      const { wrapper, stubbedActions } = mount(GroupAndRoomInputSection);

      wrapper.find({ ref: 'roomNameInput' }).element.value = 'my-new-room';
      wrapper.find({ ref: 'roomNameInput' }).trigger('input');

      expect(stubbedActions.createRoom.setRoomName).toHaveBeenCalledWith(
        expect.anything(),
        'my-new-room',
        undefined
      );
    });
  });
});
