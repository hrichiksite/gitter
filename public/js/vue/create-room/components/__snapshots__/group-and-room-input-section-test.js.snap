// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupAndRoomInputSection errors matches snapshot 1`] = `
<section>
  <div
    class="name-input-section"
  >
    <div
      class="group-select-section"
    >
      <h3
        class="section-heading"
      >
        <label>
          Community
        </label>
      </h3>
       
      <select-dropdown-stub
        error="Error: groupsRequest error"
        filterplaceholder="Filter communities"
        filtertext=""
        items=""
      />
    </div>
     
    <div
      class="room-name-section"
    >
      <h3
        class="section-heading"
      >
        <label
          for="create-room-name-input"
        >
          Room name
        </label>
      </h3>
       
      <input
        class="create-room-name-input"
        id="create-room-name-input"
        pattern="[a-zA-Z0-9\\\\-_\\\\.]+"
        placeholder="my-awesome-room"
        required="required"
      />
    </div>
  </div>
   
  <div
    class="name-input-error-area"
  >
    
    group error
    room name error
    
    <span>
      
      Error occured while fetching your communities: Error: groupsRequest error
    
    </span>
     
    <span>
      
      Error occured while fetching your repos: Error: reposRequest error
    
    </span>
  </div>
</section>
`;

exports[`GroupAndRoomInputSection matches snapshot 1`] = `
<section>
  <div
    class="name-input-section"
  >
    <div
      class="group-select-section"
    >
      <h3
        class="section-heading"
      >
        <label>
          Community
        </label>
      </h3>
       
      <select-dropdown-stub
        filterplaceholder="Filter communities"
        filtertext=""
        items=""
      />
    </div>
     
    <div
      class="room-name-section"
    >
      <h3
        class="section-heading"
      >
        <label
          for="create-room-name-input"
        >
          Room name
        </label>
      </h3>
       
      <input
        class="create-room-name-input"
        id="create-room-name-input"
        pattern="[a-zA-Z0-9\\\\-_\\\\.]+"
        placeholder="my-awesome-room"
        required="required"
      />
    </div>
  </div>
   
  <div
    class="name-input-error-area"
  >
    
    
    
    
    <!---->
     
    <!---->
  </div>
</section>
`;
